### PR TITLE
Feature - Preliminary BART screen + custom Balloon composable

### PR DESCRIPTION
**Pull request by TaxiKab119**
The BalloonRatio.kt file was used to make the ratios for the balloon available for various `Box()` sizes as the original canvas drawing was created using a 300.dp box. I am 98% sure the Balloon.kt file will need to be changes to make the code more readable/efficient but I wanted to save some _workable_ progress before moving on.

The navigation setup is crude and doesn't include all the options for categories on the landing screen (which at the moment simply log what button was clicked). The `LandingPageContent()` composable and `NavHost()` / `NavGraphBuilder` will have to be updated accordingly later.

I should also mention that the LockScreenOrientation.kt file was taken from some Google searching and a StackOverflow response. I am not super confident about what is happening in that code block so any suggestions/explanation would be valued.

### CODE REVIEW

**Comment by TonyKazanjian** [comment]
[Create a preliminary screen for the BART test](https://trello.com/c/ekxKb7vm/3-create-a-preliminary-screen-for-the-bart-test)

File: app/src/main/java/com/example/dancognitionapp/bart/Balloon.kt [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+fun getOffsetXCoordinate(x: Int, canvasSize: Int): Int {
+    return ((canvasSize / 2.0) - (x / 2.0)).roundToInt()
+}
+
+fun getOffsetYCoordinate(
+    y: Int,
+    canvasSize: Int,
+    ovalNumber: Int,
+    middleOvalHeight: Int
+): Int {
```
Comment (TonyKazanjian) [inline app/src/main/java/com/example/dancognitionapp/bart/Balloon.kt:179]: See above comment

File: app/src/main/java/com/example/dancognitionapp/bart/Balloon.kt [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+                    useCenter = false,
+                    startAngle = -55f,
+                    sweepAngle = 290f
+                )
+            }
+    )
+}
+
+fun getOffsetXCoordinate(x: Int, canvasSize: Int): Int {
+    return ((canvasSize / 2.0) - (x / 2.0)).roundToInt()
```
Comment (TonyKazanjian) [inline app/src/main/java/com/example/dancognitionapp/bart/Balloon.kt:171]: Have this return a Float, and you won't have to convert to dp and then to px

**Review by TonyKazanjian — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/dancognitionapp/bart/BartTestScreen.kt [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+import com.example.dancognitionapp.ui.LockScreenOrientation
+import com.example.dancognitionapp.ui.theme.DanCognitionAppTheme
+
+@Composable
+fun BartScreen(
+    modifier: Modifier = Modifier
+) {
+    LockScreenOrientation(orientation = ActivityInfo.SCREEN_ORIENTATION_LANDSCAPE)
+
+    Column(
```
Comment (TonyKazanjian) [inline app/src/main/java/com/example/dancognitionapp/bart/BartTestScreen.kt:41]: As we discussed, let's rewrite this in a ConstraintLayout

File: app/src/main/java/com/example/dancognitionapp/MainActivity.kt [hunk 4]
```diff
@@ -20,7 +21,7 @@ class MainActivity : ComponentActivity() {
                     modifier = Modifier.fillMaxSize(),
                     color = MaterialTheme.colorScheme.background
                 ) {
-                    LandingPageScreen()
+                    DanCognitionApp()
```
Comment (TonyKazanjian) [inline app/src/main/java/com/example/dancognitionapp/MainActivity.kt:24]: Go back to LandingScreen, remove `DanCognitionApp()`

File: app/src/main/java/com/example/dancognitionapp/bart/Balloon.kt [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+
+    val inflatorOffsetX = getOffsetXCoordinate(inflatorWidth, boxSize)
+    val inflatorOffsetY = getOffsetYCoordinate(
+        bottomOvalWidth,
+        boxSize,
+        4,
+        middleOvalHeight
+    )
+
+    Box(
```
[the remark this answers is not in the document]
Comment (TaxiKab119) [reply app/src/main/java/com/example/dancognitionapp/bart/Balloon.kt]: Resolved from out conversation on Tuesday! I'll push a new Balloon shortly.

File: app/src/main/java/com/example/dancognitionapp/bart/BartTestScreen.kt [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+            .padding(12.dp, vertical = 24.dp)
+    ) {
+        Text(
+            text = stringResource(id = contentsId),
+            style = MaterialTheme.typography.displayMedium
+        )
+    }
+}
+
+@Preview(
```
[the remark this answers is not in the document]
Comment (TaxiKab119) [reply app/src/main/java/com/example/dancognitionapp/bart/BartTestScreen.kt:113]: Sweet that's much better!

File: app/src/main/java/com/example/dancognitionapp/bart/Balloon.kt (hunk 2 above)
Comment (TaxiKab119) [reply app/src/main/java/com/example/dancognitionapp/bart/Balloon.kt:173]: I've now messed around with Float values in the new implementation of Balloon! You're totally right Float is easier...

File: app/src/main/java/com/example/dancognitionapp/ui/LandingPageScreen.kt [hunk 7]
```diff
@@ -33,21 +34,25 @@ import timber.log.Timber
 
 @OptIn(ExperimentalMaterial3Api::class)
 @Composable
-fun LandingPageScreen(modifier: Modifier = Modifier) {
+fun LandingPageScreen(onClick: (Int) -> Unit, modifier: Modifier = Modifier) {
```
[the remark this answers is not in the document]
Comment (TaxiKab119) [reply app/src/main/java/com/example/dancognitionapp/ui/LandingPageScreen.kt:37]: Sweet! I think I understand why "Modifier should be the first optional parameter in the parameter list" if you're using using the trailing lambda syntax. I will double check for this in my next push to try to not have this issue come up again!

File: app/src/main/java/com/example/dancognitionapp/ui/LockScreenOrientation.kt [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.DisposableEffect
+import androidx.compose.ui.platform.LocalContext
+
+/**
+ * https://stackoverflow.com/questions/69230049/how-to-force-orientation-for-some-screens-in-jetpack-compose
+ * */
+
+@Composable
+fun LockScreenOrientation(orientation: Int) {
```
[the remark this answers is not in the document]
Comment (TaxiKab119) [reply app/src/main/java/com/example/dancognitionapp/ui/LockScreenOrientation.kt:15]: Thanks so much for working on this! I'm currently reading through your PR now and I'm looking up things as I go to try to get the best understanding that I can!

**Pull request merged**